### PR TITLE
DM-39944: Update Execution butler builder to use Butler._registry

### PR DESCRIPTION
**Pull request by andy-slac**
Depends on lsst/daf_butler#864

## Checklist

- [X] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/pipe/base/executionButlerBuilder.py [hunk 1]
```diff
@@ -240,7 +240,7 @@ def _export(butler: Butler, collections: Iterable[str] | None, inserts: DataSetT
     # export/import
     BackendClass = get_class_of(butler._config["repo_transfer_formats", "yaml", "export"])
     backend = BackendClass(yamlBuffer, universe=butler.dimensions)
-    exporter = RepoExportContext(butler.registry, butler._datastore, backend, directory=None, transfer=None)
+    exporter = RepoExportContext(butler._registry, butler._datastore, backend, directory=None, transfer=None)
```
Comment (timj) [inline python/lsst/pipe/base/executionButlerBuilder.py:243]: It is looking suspiciously like `RepoExportContext` is not a public API...

**Comment by timj** [comment]
Sorry about the ruff problem. Not sure how I committed such a broken noqa statement last time.

File: python/lsst/pipe/base/executionButlerBuilder.py (hunk 1 above)
Comment (andy-slac) [reply python/lsst/pipe/base/executionButlerBuilder.py:243]: I think its instantiation is intended to happen inside `Butler.export()`, but execution butler does not like to use export() for some reason. I prefer to wait until we remove execution butler rater than fixing it.

Comment (TallJimbo) [reply python/lsst/pipe/base/executionButlerBuilder.py:243]: I consider everything but construction of `RepoExportContext` to be public API, but construction definitely is not.

**Review by TallJimbo — APPROVED** [review]

**Pull request merged**